t/cli.t

```diff
@@ -58,71 +62,108 @@ subtest 'local-config' => sub {
 
-        is test('generate-config', 'foo', $path), '', 'Nothing to STDOUT';
+        is test( [ 'generate-config', 'foo', $path ] ), '',
+            'Nothing to STDOUT';
 
         my $out = do {
             open my $fh, '<', $path or die;
             local $/, <$fh>;
         };
 
-        like $out, qr/^# This file configures the unofficial New Relic Perl Agent/m,
+        like $out,
+            qr/^# This file configures the unofficial New Relic Perl Agent/m,
             'Generates a config file';
-        like $out, qr/^license_key: 'foo'/m, 'Prints config with new license key';
+        like $out, qr/^license_key: 'foo'/m,
+            'Prints config with new license key';
     };
 };
 
 subtest 'run-perl' => sub {
+    # this assumes NewRelic is enabled
+    local $ENV{NEWRELIC_ENABLED}     = 1;
+    local $ENV{NEWRELIC_CONFIG_FILE} = dist_file 'NewFangle-Agent',
+        'test.yml';
+    local $ENV{NEWRELIC_ENVIRONMENT} = 'environment';
+    local $ENV{NEWRELIC_DAEMON_HOST} = '127.0.0.1:8000';
+    local $ENV{NEWRELIC_APP_NAME}    = 'unit tests';
+
     require NewFangle::Agent;
-    is test(qw( run-perl -Ilib -E ), 'say ">> $NewFangle::Agent::VERSION <<"'),
+    is test(
+        [ qw( run-perl -Ilib -E ), 'say ">> $NewFangle::Agent::VERSION <<"' ],
+        qr/failed to connect to daemon/,
+        ),
         ">> $NewFangle::Agent::VERSION <<\n",
         'Loaded NewFangle::Agent when running Perl';
+
+    is test(
+        [ qw( run-perl --tx wrapped-job -E ), 'say ">> HELLO <<"' ],
+        qr/failed to connect to daemon/,
+        ),
+        ">> HELLO <<\n",
+        'Perl expression successfully run with the transaction wrapper';
+
+    is test( [qw( run-perl --tx wrapped-job any-old-rubbish )],
+        qr/failed to connect to daemon/, 512, ),
+        "",
+        'Perl script error is captured when run with the transaction wrapper';
+
+    like test( [qw( run-perl --tx wrapped-job )], "", 256 ),
+        qr/^Usage: newrelic-admin run-perl/,
+        'Help page is output when no command is specifed';
 };
 
 subtest 'version' => sub {
     subtest 'Command' => sub {
-        like test('version'), qr/newrelic-admin v/, 'Print version';
+        like test( ['version'] ), qr/newrelic-admin v/, 'Print version';
     };
 
     subtest 'Option' => sub {
-        like test('--version'), qr/newrelic-admin v/, 'Print version';
+        like test( ['--version'] ), qr/newrelic-admin v/, 'Print version';
     };
 };
 
 done_testing;
 
-sub test {
-    my @args = @_;
-    my ( $out, $err, $ret ) = capture { system 'bin/newrelic-admin', @args };
+sub test ( $args, $exp_err = "", $exp_ret = 0 ) {
+    my ( $out, $err, $ret )
+        = capture { system 'bin/newrelic-admin', $args->@* };
 
-    is $err, '', 'Nothing on STDERR';
-    is $ret, 0,  'Exits without error';
+    if ( !$exp_err ) {
+        is $err, '', 'Nothing on STDERR';
+    }
+    if ( !$exp_ret ) {
+        is $ret, 0, 'Exits without error';
+    }
```

Comment:
Why aren't we testing that `$err` and `$ret` are what the user asked for, if they ask for something?